http-client-core/src/main/java/io/micronaut/http/client/exceptions/HttpClientException.java

```diff
@@ -50,5 +52,26 @@ public HttpClientException(String message, Throwable cause, boolean shared) {
         if (!shared) {
             throw new IllegalArgumentException("shared must be true");
         }
+        serviceIdLocked = true;
+    }
+
+    public final String getServiceId() {
+        return serviceId;
+    }
```

Comment:
@yawkat 

Can you the return type with `@Nullabe` 

```java
 * @since 3.6.0
 */
@Nullable
public final String getServiceId() {
        return serviceId;
    }
```

or 

```java
 * @since 3.6.0
 */
@NonNull
public final Optional<String> getServiceId() {
        return Optional.ofNullable(serviceId);
    }

```